### PR TITLE
[RLlib] Add support for reading from directories with dataset

### PR DESCRIPTION
**Pull request by avnishn**
Signed-off-by: avnish <avnish@anyscale.com>

This should solve the RLlib related issues in #26410 

CC @jiaodong 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jiaodong** [comment]
thanks for the quick fix ! Can you attached run results with this PR on the offline+online rl example ? Given this API is rllib related i will defer to them for final approval.

**Comment by richardliaw** [comment]
We should enable the rl_offline and rl_online tests as part of this PR.

**Comment by jiaodong** [comment]
looks like examples:rl_offline_example and examples:rl_online_example are still failing on CI of latest commit ?

**Comment by richardliaw** [comment]
is it really a rllib issue or just a predictor issue?

**Comment by jiaodong** [comment]
I think it's mostly AIR issue, but the way this notebook runs uses a few modules that live in rllib folder that seems to diverge from AIR's implementation a bit. I didn't read all files however, need more sleep today =.=

**Comment by jiaodong** [comment]
lg but you'll need a RLlib code owner to take another look

File: rllib/offline/dataset_reader.py [hunk 1]
```diff
@@ -142,7 +142,16 @@ def get_dataset_and_shards(
...
+            inputs = Path(paths).absolute()
+            if inputs.is_dir():
+                if format == "json":
+                    paths = list(inputs.glob("*.json")) + list(inputs.glob("*.zip"))
+                else:
+                    # we don't support zip files of parquet
+                    paths = list(inputs.glob("*.parquet"))
+                paths = [str(path) for path in paths]
+            else:
+                paths = [paths]
```
Comment (richardliaw) [inline rllib/offline/dataset_reader.py:154]: why do we need to do this? doesn't datasets automatically glob the paths?

Comment (avnishn) [reply rllib/offline/dataset_reader.py:154]: does it! Its not mentioned in the documentation so I wasn't aware of that. I'll try that out sometime tomorrow

Comment (avnishn) [reply rllib/offline/dataset_reader.py]: ok I can confirm this is the case... that being said should we still merge the test or is it not worth it?

Comment (avnishn) [reply rllib/offline/dataset_reader.py]: gonna close this for now

**Pull request closed without merging**